mtags/src/main/scala-2/scala/meta/internal/pc/PcInlineValueProviderImpl.scala

```diff
@@ -66,8 +66,18 @@ final class PcInlineValueProviderImpl(
   }
 
   private def symbolsUsedInDef(
+      symbol: Symbol,
       rhs: Tree
   ): List[Symbol] = {
+    val classParents =
+      if (symbol != null) symbol.ownersIterator.filter(_.isType).toSet
+      else Set.empty
+
+    def validateSymbol(symbol: Symbol) =
+      !symbol.isSynthetic && !symbol.isImplicit &&
+        (!symbol.safeOwner.isType || symbol.safeOwner.isModuleClass || classParents
```

Comment:
What about:
```
class O {
  val t = new T()
  import t._
  val bb = a + a

  class Inner {
    val a = 123
    val cc = <<bb>>
  }
}
```
both a's have a type as parent. We should differentiate Select and Ident. I think the logic is correct on Select, but not on Ident

Reply:
I found another example, where this logic fails but in a different way.
```scala
object O {
  def ==(o: O) = false
}
object P {
  def test() = {
   val isOne = O == O
    <<i>>sOne //claims it cannot inline
  }
}
```
Being concerned only with the first qualifier for select (`this` might need some special logic) should be a better approach. Ident was for sake of `infix` methods, I guess we should detect them and handle appropriately.

However, this will fail to cover overshadow connected to extension methods and implicit classes. But lets maybe make it into a followup issue.

Reply:
> However, this will fail to cover overshadow connected to extension methods and implicit classes. But lets maybe make it into a followup issue.

Yeah, we should be able to check that separately.

Reply:
That's not great then :thinking: Maybe we can check the ident span in Select(ident: Ident, sel)  to see if it actually exist in the code ?

Reply:
I changed the approach and I check if there are in scope before inline and look for conflicts on these symbols. Seems like the simplest solution.